homeassistant/components/assist_satellite/__init__.py

```diff
@@ -0,0 +1,195 @@
+            }
+        ),
+        func=async_service_get_text,
+        required_features=[AssistSatelliteEntityFeature.TRIGGER],
+        supports_response=SupportsResponse.OPTIONAL,
+    )
+
+    component.async_register_entity_service(
+        name=SERVICE_SAY_TEXT,
+        schema=cv.make_entity_service_schema(
+            {vol.Required(ATTR_ANNOUNCE_TEXT): cv.string}
+        ),
+        func=async_service_say_text,
+        required_features=[AssistSatelliteEntityFeature.TRIGGER],
+        supports_response=SupportsResponse.NONE,
+    )
+
+    component.async_register_entity_service(
+        name=SERVICE_MUTE_MICROPHONE,
+        schema=cv.make_entity_service_schema({vol.Required(ATTR_MUTED): cv.boolean}),
+        func=async_service_mute_microphone,
+        required_features=[AssistSatelliteEntityFeature.AUDIO_INPUT],
+        supports_response=SupportsResponse.NONE,
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent[AssistSatelliteEntity] = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+async def async_service_wait_wake(
+    entity: AssistSatelliteEntity, service_call: ServiceCall
+) -> str | None:
+    """Wait for one or more wake words to be spoken."""
+    if announce_text := service_call.data.get(ATTR_ANNOUNCE_TEXT):
+        # Announce first
+        await entity.async_run_pipeline_on_satellite(
+            start_stage=assist_pipeline.PipelineStage.TTS,
+            end_stage=assist_pipeline.PipelineStage.TTS,
+            run_config=PipelineRunConfig(announce_text=announce_text),
+        )
+
+    # Wait for wake word(s)
+    result = await entity.async_run_pipeline_on_satellite(
+        start_stage=assist_pipeline.PipelineStage.WAKE_WORD,
+        end_stage=assist_pipeline.PipelineStage.WAKE_WORD,
+        run_config=PipelineRunConfig(
+            wake_word_names=service_call.data.get(ATTR_WAKE_WORDS)
+        ),
+    )
+
+    if (not service_call.return_response) or (result is None):
+        return None
+
+    text = result.detected_wake_word or ""
+    return text.strip()
+
+
+async def async_service_get_text(
+    entity: AssistSatelliteEntity, service_call: ServiceCall
+) -> str | None:
+    """Wait for a response from the user."""
+    if announce_text := service_call.data.get(ATTR_ANNOUNCE_TEXT):
+        # Announce first
+        await entity.async_run_pipeline_on_satellite(
+            start_stage=assist_pipeline.PipelineStage.TTS,
+            end_stage=assist_pipeline.PipelineStage.TTS,
+            run_config=PipelineRunConfig(announce_text=announce_text),
+        )
+
+    if service_call.data.get(ATTR_PROCESS):
```

Comment:
For a future PR: When it's being processed, especially by LLMs, we should be able to pass some extra context for the conversation agent to understand the response of the user that comes in (ie, extra context would be the entity ID of the garage door that was left open)